### DISCUSSION TITLE
Error with Custom Spatially Varying Elastic Moduli Material

### DISCUSSION THREAD

**Discussion by josebastiase** [idaholab/moose#27320] [Q&A Modules: Porous Flow] [answered]
Hi experts,

I'm encountering an error while running an input file for my custom material. The material computes a spatially varying elastic tensor. Not sure what is going on...

Please find the material files, a minimal example, and log files in debug mode.

[Files
](https://github.com/josebastiase/Tensor_mech/tree/main)

**Comment by GiudGiud** [comment]
Hello

my guess without having run your code is that the residualSetup routine for the derived class needs to call the residualSetup for the parent class first. 

If it s not that, please run your input in a debugger and get us the exact line for the "out of bounds" error that you receive currently. 

Guillaume

**Reply by josebastiase** [reply]
Thank you! So, I added a line in the constructor to call the residualSetup function of the base class ComputeElasticityTensorBase:

`template <bool is_ad, typename T>
void
IsotropicElasticModulusFromVarTempl<is_ad, T>::residualSetup()
{

 // Call residualSetup of base class
 ComputeElasticityTensorBaseTempl<is_ad, T>::residualSetup();`

However, this didn't work. The backtrace of the debugger is in the file named backtrace.txt, accessible through this [link](https://github.com/josebastiase/Tensor_mech)

**Reply by GiudGiud** [reply]
can you update `IsotropicElasticModulusFromVar.C` on your repo? The backtrace points to a wrong line currently. 

You should check:
- which material properties are being accessed in your material
- which material properties are resized in `ComputeElasticityTensorBaseTempl`

**Reply by josebastiase** [reply]
Im sorry, should be updated now. Could it `MooseArray<double>::operator[]` the issue?

**Reply by GiudGiud** [reply]
That operator is indexing into the array of the material properties at the index of the quadrature point. And for some reason, that material property has not been initialized, the size is too small

**Reply by GiudGiud** [reply]
ok nevermind.

it looks to me like the seg fault is happening when accessing the variable values.

**Reply by GiudGiud** [reply, marked as answer]
so with that in mind, is residualSetup the right place for what you are doing? 

Residual setup is not in a quadrature point loop maybe?
I think you may be wanting to override another material property routine. 
Is _cijk supposed to depend on space? Isnt it a constant in the base class?

**Reply by josebastiase** [reply]
Hi @GiudGiud 

Thank you! Yeah the issue was that I didnt need to override residualSetup. I removed it and it worked ;)
